### ISSUE TITLE
Build fails on JDK 16

### ISSUE THREAD

**Issue by eeverman** [eeverman/andhow#566]
The build works when using JDKs 1.8, 11, & 15, however there are test errors and the build fails when using JDK 16.

It looks like there are two different types of errors:
```
[INFO] Running org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.007 s <<< FAILURE! - in org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest
[ERROR] org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest.completeWorkflow  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.creation.cglib.ClassImposterizer$3
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest.setUp(KillAndHowBeforeAllTestsExtTest.java:54)
```
and
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.17 s <<< FAILURE! - in org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest
[ERROR] org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.completeWorkflow  Time elapsed: 0.139 s  <<< ERROR!
java.lang.ExceptionInInitializerError
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.setUp(KillAndHowBeforeEachTestExtensionTest.java:50)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @3f0aaff5
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.setUp(KillAndHowBeforeEachTestExtensionTest.java:50)
```

Is there a simple fix for this or does it require a larger effort??

To compile with JDK16, an 'enforcer' rule in the root pom file needs to be changed:
```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-enforcer-plugin</artifactId>
  ...
  <version>[1.8.0,17)</version> (was 16)
  ...
</plugin>
```
This rule is there because it was known that JDK16 wasn't working, but I didn't know why.  it would be nice to figure that out so the build can happen on the latest JDK and this rule can be opened as above.

My Environment:
OS:  Mac 11.3.1 w/ M1 Processor
Java: openjdk version "16.0.1" 2021-04-20 & openjdk version "16.0.2" 2021-07-20
Maven: 3.8.1 & 3.8.2

**Comment by eeverman** [comment]
Nice find on those docs.  This project will need to deal with those same issues to support modules soon.

**Comment by eeverman** [comment]
Fixed by #573
